e2e/src/tests.rs

```diff
@@ -0,0 +1,295 @@
+
+#[allow(dead_code)]
+// target/debug/mirrord exec --pod-name pod_name  -c binary command
+fn start_server(pod_name: &str, command: Vec<&str>, env: HashMap<&str, &str>) -> Child {
+    let path = format!(
+        "{}/../target/debug/mirrord",
+        env::current_dir().unwrap().display()
+    );
+    println!("{}", path);
+    let args: Vec<&str> = vec!["exec", "--pod-name", pod_name, "-c"]
+        .into_iter()
+        .chain(command.into_iter())
+        .collect();
+    println!("{:?}", args);
+    let server = Command::new(path)
+        .args(args)       
+        .env("MIRRORD_AGENT_NAMESPACE", "test-namespace")
+        .envs(&env)
+        .stdout(Stdio::piped())
+        .stderr(Stdio::piped())
+        .spawn()
+        .unwrap();
+    server
+}
+
+// (minikube service nginx --url)
+#[allow(dead_code)]
+fn get_service_url(namespace: &str) -> String {
+    let cmd = Command::new("minikube")
+        .args(["service", "nginx", "--namespace", namespace, "--url"])
+        .output()
+        .unwrap();
+    let url = String::from_utf8(cmd.stdout).unwrap();
+    // output contains a left trailing newline
+    url.trim_matches(&['\r', '\n'][..]).to_string()
+}
+
+// kubectl get pods -o custom-columns=:metadata.name | grep nginx
+#[allow(dead_code)]
+fn get_nginx_pod_name(namespace: &str) -> String {
+    // Todo: we can use the kubernetes API here
+    let cmd = Command::new("kubectl")
+        .args([
+            "get",
+            "pods",
+            "--namespace",
+            namespace,
+            "-o",
+            "custom-columns=:metadata.name",
+            "--selector=app=nginx",
+        ])
+        .output()
+        .unwrap();
+    let pod_name = String::from_utf8(cmd.stdout).unwrap();
+    pod_name.trim_matches(&['\r', '\n'][..]).to_string()
+}
+
+#[allow(dead_code)]
+fn create_namespace(namespace: &str) {
+    let cmd = Command::new("kubectl")
+        .args(["create", "namespace", namespace])
+        .output()
+        .unwrap();
+}
+
+#[allow(dead_code)]
+fn delete_namespace(namespace: &str) {
+    let cmd = Command::new("kubectl")
+        .args(["delete", "namespace", namespace])
+        .output()
+        .unwrap();
+}
+
+#[allow(dead_code)]
+fn create_nginx_pod(namespace: &str) {
+    let cmd = Command::new("kubectl")
+        .args(["apply", "--namespace", namespace, "-f", "app.yaml"])
+        .spawn()
+        .unwrap();    
+}
+
+
+// replicate curl behavior
+#[allow(dead_code)]
+fn curl_request(url: &str, method: &str) {
+    // we can use the reqwest/hyper library here
+    Command::new("curl")
+        .args(["-X", method, "--data", TEXT, url])
+        .output()
+        .unwrap();
+}
+
+#[allow(dead_code)]
+fn validate_data(read: String, service_url: &str, server: &mut Child) {
```

Comment:
I don't like this function.. It's flow is very odd and weird. I suggest it'll get the stdout reader, server and service url
and then it would do something like
```rust
request(service_url, "GET")
assert_eq!(stdout.read.., "GET: Request completed"
```